1a

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [11]:
!find "/content/drive/MyDrive" -name "parsed.csv"

/content/drive/MyDrive/MSSP607/Modules/Week_4/Files_export/Working with Pandas DataFrames/parsed.csv
/content/drive/MyDrive/MSSP607/Modules/Week_4/Files_export/parsed.csv


In [13]:
import pandas as pd

df = pd.read_csv(
    '/content/drive/MyDrive/MSSP607/Modules/Week_4/Files_export/parsed.csv'
)

df.head()

,alert,cdi,code,detail,dmin,felt,gap,ids,mag,magType,...,status,time,title,tsunami,type,types,tz,updated,url,parsed_place
0,NaN,NaN,37389218,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.008693,NaN,85.0,",ci37389218,",1.35,ml,...,automatic,1539475168010,"M 1.4 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.0,1539475395144,https://earthquake.usgs.gov/earthquakes/eventp...,California
1,NaN,NaN,37389202,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.020030,NaN,79.0,",ci37389202,",1.29,ml,...,automatic,1539475129610,"M 1.3 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.0,1539475253925,https://earthquake.usgs.gov/earthquakes/eventp...,California
2,NaN,4.4,37389194,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.021370,28.0,21.0,",ci37389194,",3.42,ml,...,automatic,1539475062610,"M 3.4 - 8km NE of Aguanga, CA",0,earthquake,",dyfi,focal-mechanism,geoserve,nearby-cities,o...",-480.0,1539536756176,https://earthquake.usgs.gov/earthquakes/eventp...,California
3,NaN,NaN,37389186,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.026180,NaN,39.0,",ci37389186,",0.44,ml,...,automatic,1539474978070,"M 0.4 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.0,1539475196167,https://earthquake.usgs.gov/earthquakes/eventp...,California
4,NaN,NaN,73096941,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.077990,NaN,192.0,",nc73096941,",2.16,md,...,automatic,1539474716050,"M 2.2 - 10km NW of Avenal, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,scit...",-480.0,1539477547926,https://earthquake.usgs.gov/earthquakes/eventp...,California


In [14]:
df.columns

Index(['alert', 'cdi', 'code', 'detail', 'dmin', 'felt', 'gap', 'ids', 'mag',
       'magType', 'mmi', 'net', 'nst', 'place', 'rms', 'sig', 'sources',
       'status', 'time', 'title', 'tsunami', 'type', 'types', 'tz', 'updated',
       'url', 'parsed_place'],
      dtype='object')

2a

In [15]:
japan_mb = df[
    (df['parsed_place'].str.endswith('Japan')) &
    (df['magType'] == 'mb')
]

japan_mb['mag'].quantile(0.95)

np.float64(4.9)

The 95th percentile of earthquake magnitude in Japan for mb earthquakes is 4.9. This means 95% of these earthquakes had magnitudes at or below 4.9.

2b

In [16]:
indonesia = df[
    df['parsed_place'].str.endswith('Indonesia')
]

indonesia['tsunami'].value_counts(normalize=True)

,proportion
tsunami,
0,0.768707
1,0.231293


In [17]:
tsunami_percentage = indonesia['tsunami'].mean()

print(f"Percentage of earthquakes in Indonesia with tsunamis: {tsunami_percentage:.2%}")

Percentage of earthquakes in Indonesia with tsunamis: 23.13%


About 23.13% of earthquakes in Indonesia were associated with tsunamis. This means roughly one out of every four earthquakes in the dataset had a tsunami.

2c

In [18]:
nevada = df[
    df['parsed_place'].str.endswith('Nevada')
]

nevada.describe()

,cdi,dmin,felt,gap,mag,mmi,nst,rms,sig,time,tsunami,tz,updated
count,15.000000,681.000000,15.000000,681.000000,681.000000,1.00,681.000000,681.000000,681.000000,6.810000e+02,681.0,681.0,6.810000e+02
mean,2.440000,0.166199,2.400000,153.668120,0.500073,2.84,12.618209,0.151986,10.970631,1.538314e+12,0.0,-480.0,1.538402e+12
std,0.501142,0.166228,4.626013,68.735302,0.696710,NaN,9.866963,0.084662,19.607150,5.965637e+08,0.0,0.0,6.010951e+08
min,2.000000,0.001000,1.000000,29.140000,-0.500000,2.84,3.000000,0.000500,0.000000,1.537247e+12,0.0,-480.0,1.537307e+12
25%,2.000000,0.053000,1.000000,97.380000,-0.100000,2.84,6.000000,0.106900,0.000000,1.537854e+12,0.0,-480.0,1.537928e+12
50%,2.200000,0.112000,1.000000,149.140000,0.400000,2.84,10.000000,0.146300,2.000000,1.538280e+12,0.0,-480.0,1.538428e+12
75%,2.900000,0.233000,1.000000,199.720000,0.900000,2.84,16.000000,0.187100,12.000000,1.538821e+12,0.0,-480.0,1.538878e+12
max,3.300000,1.414000,19.000000,355.910000,2.900000,2.84,61.000000,0.863400,129.000000,1.539461e+12,0.0,-480.0,1.539483e+12


There were 681 earthquake records in Nevada. The average magnitude was about 0.50, and the maximum magnitude was 2.90.

2d

In [19]:
df['ring_of_fire'] = df.parsed_place.str.contains(r'|'.join([
    'Bolivia', 'Chile', 'Ecuador', 'Peru', 'Costa Rica',
    'Guatemala', '^Mexico', 'Japan', 'Philippines', 'Indonesia',
    'New Zealand', 'Antarctic', 'Canada', 'Fiji',
    'Alaska', 'Washington', 'California', 'Russia',
    'Taiwan', 'Tonga', 'Kermadec Islands'
]))

df[['ring_of_fire', 'parsed_place']]

,ring_of_fire,parsed_place
0,True,California
1,True,California
2,True,California
3,True,California
4,True,California
...,...,...
9327,True,California
9328,True,California
9329,False,Puerto Rico
9330,True,California


2e

In [20]:
earthquakes = df[df['type'] == 'earthquake']

earthquakes['ring_of_fire'].value_counts()

,count
ring_of_fire,
True,7004
False,2077


There were 7,004 earthquakes in Ring of Fire locations and 2,077 earthquakes outside the Ring of Fire.

2f

In [21]:
tsunami_count = df.loc[df['ring_of_fire'], 'tsunami'].sum()

print("Tsunami count along the Ring of Fire:", tsunami_count)

Tsunami count along the Ring of Fire: 45


There were 45 tsunamis along the Ring of Fire.